sphinx/texinputs/sphinxlatexadmonitions.sty

```diff
@@ -51,78 +59,59 @@
        \rule[\dimexpr.4\baselineskip-\spx@notice@border\relax]
              {\linewidth}{\spx@notice@border}}\hss}\allowbreak
   }% end of sphinxlightbox environment definition
-% may be renewenvironment'd by user for complete customization
+
+% note/hint/important/tip notices
+%
+% Since 1.5 these environments are named individually to allow user to
+% redefine them entirely.
+%
+% The Sphinx definitions were done like this, prior to 6.2.0:
+%
+% \newenvironment{sphinxhint}[1]
+%   {\begin{sphinxlightbox}\sphinxstrong{#1} }{\end{sphinxlightbox}}
+%
+% It was possible for user to \renewenvironment this to use sphinxheavybox,
+% but this had stopped functioning at 5.1.0 which had greatly enhanced the
+% sphinxheavybox environment and the needed support macros existed only for
+% warning, caution, attention, danger, and error admonitions.  At 6.1.0 a
+% patch was added to allow the \renewenvironment method to work again for
+% note, hint, important, tip to use sphinxheavybox.
+%
+% Then at 6.2.0, note, hint, important, tip can now access full customization
+% of sphinxheavybox: CSS-named alike options (if used) trigger sphinxheavybox.
+% The \renewenvironment method still works but then only common border width,
+% non-configurable padding, straight corners and no shadow are all implied,
+% as long as the related CSS-named options are not used.
 \newenvironment{sphinxnote}[1]
-  {\begin{sphinxlightbox}\sphinxstrong{#1} }{\end{sphinxlightbox}}
+  {\edef\spx@env{sphinx\ifspx@opt@heavynote heavy\else light\fi box}%
+   \expanded{\noexpand\begin{\spx@env}}\sphinxstrong{#1} }
```

Comment:
This ``\expanded`` requires a pdflatex from 2019 at most.  I will need to revert to older LaTeX syntax.